### ISSUE TITLE
ENH: stats: add array API-support

### ISSUE THREAD

**Issue by mdhaber** [scipy/scipy#20544]
This issue tracks progress toward the addition of array-API support to `scipy.stats` functions. The functions listed below look ready for conversion, and I'd be happy to review PRs for them. Priority, balancing the ease and importance of the task, is roughly in the order listed. 

- [x] `moment` (gh-20292)
- [x] `skew` (gh-20541 - please see this PR as an example roughly similar `kurtosis` through `circstd`)
- [ ] `kurtosis`
- [ ] `describe`
- [ ] `mode`
- [ ] `variation`
- [ ] `sem`
- [ ] `circmean`
- [ ] `circvar`
- [ ] `circstd`
- [x] `pearsonr` (gh-20284)
- [ ] `ttest_1samp` (I might do this one to help pave a path toward the rest)
- [ ] `ttest_rel`
- [ ] `ttest_ind`
- [ ] `skewtest`
- [ ] `kurtosistest`
- [ ] `normaltest`

Most other functions are either more challenging (e.g. because they do not already natively support an `axis` argument) or lower priority. Let's pick the low-hanging fruit (listed above) first. If you're particularly interested in another function, I'd appreciate it if you'd comment below before starting work.

**Comment by lucascolley** [comment]
> add N-D support to _array_api.cov; consider making it public if array API won't offer it

Public _in SciPy_ seems wrong. Upstreamed into either the standard or a new library for these type of standard-extensions seems right.

**Comment by lucascolley** [comment]
Thinking a little more about `mean`, `var`, `std` etc., (when you get round to it) I think it would be useful to open an issue on the array API repo discussing the fully-featured versions as you would potentially implement them within SciPy. The question is whether or not those versions would belong on the array library level, and hence (eventually) in the standard. If the answer is:
- "yes, we just need time to get the API right / see adoption", then keeping things private in SciPy is definitely the way to go to avoid duplication in the future.
- "_maybe_ (grey area / bordering on array-consumer level)", then I would also suggest keeping things private in SciPy for now, as it may be that they would find a more appropriate home in a hypothetical (say) `array-api-extra` package (potentially making their way into the standard as the implementation matures / the API is adopted).
- "no, these features belong on the array-consumer level", then I would go ahead with adding them to `scipy.stats`, as that seems like the ideal place!
